src/oop/programs_with_multiple_classes/medium/q4/MainTest.java

```diff
@@ -0,0 +1,92 @@
+
+    @BeforeEach
+    public void setup() {
+        String wheelClassString = "oop.programs_with_multiple_classes.medium.q4." + wheelClass;
+        String chassisClassString = "oop.programs_with_multiple_classes.medium.q4." + chassisClass;
+        wheel = new ObjectTest(wheelClassString);
+        chassis = new ObjectTest(chassisClassString);
+        String modifiedWheelMessage =
+                "You have modified the provided portions of class " + wheelClass + ". Please revert them to the original state.";
+        assertTrue(wheel.hasField(varDurability, double.class), modifiedWheelMessage);
+        assertTrue(wheel.hasField(varManufacturer, String.class), modifiedWheelMessage);
+        assertTrue(wheel.hasModifier(varDurability, "private"), modifiedWheelMessage);
+        assertTrue(wheel.hasModifier(varManufacturer, "private"), modifiedWheelMessage);
+        String modifiedChassisMessage =
+                "You have modified the provided portions of class " + chassisClass + ". Please revert them to the original state.";
+        Class<?>[] classArguments = {String.class};
+        assertTrue(chassis.hasConstructor(classArguments), modifiedChassisMessage);
+        assertTrue(chassis.hasModifier(classArguments, "public"), modifiedChassisMessage);
+    }
+
+    @Test
+    public void chassisClassHasCorrectAttributes() {
+        String incorrectFieldMessage = "Your " + chassisClass + " class is missing a required field";
+        String incorrectModifierMessage = "One of your " + chassisClass + " class attributes does not have the correct modifier.";
+        assertTrue(chassis.hasField(varMaterialType, String.class), incorrectFieldMessage);
+        assertTrue(chassis.hasModifier(varMaterialType, "private"), incorrectModifierMessage);
+    }
+
+    @Test
+    public void wheelClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {double.class, String.class};
+        assertTrue(wheel.hasConstructor(classArguments),
+                "Your " + wheelClass + " constructor does not have the correct parameters.");
+        assertTrue(wheel.hasModifier(classArguments, "public"),
+                "Your " + wheelClass + " constructor does not have the correct modifiers.");
+    }
+
+    private static Stream<Arguments> wheelInputProvider() {
+        return Stream.of(
+                Arguments.of(5.5, "Milani"),
+                Arguments.of(2.3, "Ford"),
+                Arguments.of(7.7, "Nissan"),
+                Arguments.of(-3.5, "Milani"),
+                Arguments.of(-2.6, "Martin"),
+                Arguments.of(-5.4, "Harley")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("wheelInputProvider")
+    public void wheelConstructorInitializesValuesCorrectly(double d, String m) throws Throwable {
+        Object[][] arguments = {
+                {d, double.class},
+                {m, String.class}
+        };
+        Object wheelInstance = wheel.createInstance(arguments);
+        if (d >= 0) {
```

Comment:
You don't need different asserts here. You could just use the ternary operator to set the expected value for the assert instead.